modules/contracts/apps/challenge/src/handlers/query.rs

```diff
@@ -0,0 +1,61 @@
+use crate::contract::{AppResult, ChallengeApp};
+use crate::msg::{
+    ChallengeQueryMsg, ChallengeResponse, CheckInResponse, FriendsResponse, VotesResponse,
+};
+use crate::state::{CHALLENGE_FRIENDS, CHALLENGE_LIST, DAILY_CHECK_INS, VOTES};
+use cosmwasm_std::{to_binary, Binary, Deps, Env};
+
+pub fn query_handler(
+    deps: Deps,
+    _env: Env,
+    app: &ChallengeApp,
+    msg: ChallengeQueryMsg,
+) -> AppResult<Binary> {
+    match msg {
```

Comment:
For automation purposes we could add a query here that returns if the challenge failed and a pay-out is required. A cron-cat job could be triggered on that query result.

Reply:
In the current implementation, the pay-out happens automatically when `count_vote` is called, the votes are counted and if there are any false votes, `charge_penalty is called. Currently `count_vote` has to be manually called. 

One way to use croncat to automate this could be to have a `ChallengeStatus` enum for a `status` field on `ChallengeEntry`. 

Then we can add an `end_block` on `ChallengeEntry`, if we are `>=` than the `end_block` when `daily_check_in` is called, the `ChallengeEntry.status` is set to `Finished`. Then we can use the query to kick off a croncat job which calls `count_vote`

Reply:
As long as there's no way for the owner to veto a vote IMO there's probably no need for cron-cat here because you can just call `count_vote` when one person votes against. I think one of the design choices that is most important to this kind of app is the timeline design. 

- When do votes start/end?
- When does the tallying occur? (if only possible after a specific time-stamp then cron-cat can help here)
- Is there a veto period? 

From an app-design perspective I think the day-centric way of administering the challenges is an approach that doesn't go well with on-chain logic.  Instead of working with a 24h refresh period (which needs to be in sync with user's timezone) we could work with a "ticketing" system where a vote either goes to the last ticket created (if within a customizable delay) or a new ticket is created (with a set end-date for voting and a veto-period for administration) 
That would also allow for shorter or longer duration challenges as well as automation for cleanup by cron-cat after the ticket's end-time.

Reply:
I'm going to go with simple blockheight for time within the contract first, `end_block` is set on creation of challenges and can be updated after creation. I'l get these changes up with some new tests. If anything this just allows me easier iteration with the tests until the design and flow is set. 

We can change this time modelling to a timestamp `String` later.

I have an idea for when votes start/end, which is based on the new challenge.status. Also in my new changes, counting of votes, or tallying, occurs only after `challenge.status = ChalengeStatus::OverAndUncounted`. 

Once these changes are up I think it will become clearer to us the most suitable path to go.

Reply:
With my latest updates, these are the answers to your questions above:

- When do votes start/end?
  - votes start after `challenge_entry.end_block` has passed. 
  - votes end after `tally_votes` is called (this is only callable after `challenge_entry.end_block` has passed. `tally_votes` will set the `challenge_entry.status` to either `OverAndFailed` (friends won) or `OverAndCompleted` (admin won)
  
  - When does the tallying occur? (if only possible after a specific time-stamp then cron-cat can help here)
    - When `tally_votes` is called, it will tally only if the `challenge_status == ChallengeStatus::OverAndPending`. 
    
 - Is there a veto period?
   - In the current implementation, No.